README.md

```diff
@@ -1,2 +1,130 @@
+
+
+                }
+            }
+        }
+    }
+    post {
+        always {
+            updateBuildReferences(currentBuild.number, categorySelected)
+        }
+    }
+}
+
+
+def getBuildReferences(noOfBuilds) {
+    def currentReferences = []
+    if (fileExists(env.GLOBAL_FILE)) {
+        def jsonContent = readFile(env.GLOBAL_FILE)
+        currentReferences = readJSON(text: jsonContent)
+        while (currentReferences.size() > Integer.parseInt(noOfBuilds)) {
+            currentReferences.remove(0)
+        }
+        echo "Last ${noOfBuilds} references:"
+        currentReferences.each { reference ->
+            echo "Reference: ${reference.url}"
+        }
+    } else {
+        echo "No Existings build references to show"
+    }
+}
+
+def updateBuildReferences(buildNumber, category) {
+    echo "updating build reference: ${buildNumber} - ${category}"
+    def currentReferences = []
+    if (!fileExists(env.GLOBAL_FILE)) {
+        writeFile(file: env.GLOBAL_FILE, text: '[]')
+        echo "Created new file: ${env.GLOBAL_FILE}"
+    }
+
+    def jsonContent = readFile(env.GLOBAL_FILE)
+    echo "Current references - ${jsonContent}"
+
+    //def jsonSlurper = new groovy.json.JsonSlurper()
+    //currentReferences = jsonSlurper.parseText(jsonContent)
+    currentReferences = readJSON(text: jsonContent)
+
+    def newReference = [
+            buildNumber: buildNumber,
+            url: env.BUILD_URL,
+            category: category,
+            timestamp: new Date().format("yyyy-MM-dd'T'HH:mm:ss'Z'")
+    ]
+
+    currentReferences.add(newReference)
+
+    while (currentReferences.size() > Integer.parseInt(env.TOTAL_BUILD_REFERENCES_TO_KEEP)) {
+        currentReferences.remove(0)
+    }
+
+    //def jsonOutput = JsonOutput.toJson(currentReferences)
+    //writeFile(file: env.GLOBAL_FILE, text: JsonOutput.prettyPrint(jsonOutput))
+    //echo "Updated build references: ${JsonOutput.prettyPrint(jsonOutput)}"
+
+    def jsonOutput = writeJSON(returnText: true, json: currentReferences)
+    writeFile(file: env.GLOBAL_FILE, text: jsonOutput)
+    echo "Updated build references: ${jsonOutput}"
+}
+
+def getCategoryChoicesMap() {
+    def choices = []
+    def choicesMap = [:]
+    def onboardingConfig = jenkins.model.GlobalConfiguration.all().get(io.jenkins.plugins.sample.OnboardingSectionConfiguration.class)
+    if (onboardingConfig) {
+        //Check if possible to get this from step's descriptor
+        def categories = onboardingConfig.getCategoryConfig().getCategories().collect {it}
+        categories.each { category ->
+            choicesMap[category.name] = category.uuid
+        }
+    } else {
+        echo "Onboarding configuration doesn't exist."
+    }
+    return choicesMap
+}
+```
+2. Install pipeline-utility-steps
+3. Approve method invocations from http://localhost:8080/jenkins/manage/scriptApproval/
```

Comment:
If you ever have to approve methods on this page you are probably going down the wrong path. `getBuildReferences` and `updateBuildReferences` don't look like they belong in a Pipeline at all. I think the expectation was for this logic to be done internally in `OnboardingTaskBuilder`.